instrumentation-genai/opentelemetry-instrumentation-cohere-v2/src/opentelemetry/instrumentation/cohere_v2/utils.py

```diff
@@ -0,0 +1,199 @@
+    operation_name=GenAIAttributes.GenAiOperationNameValues.CHAT.value,
+):
+    attributes = {
+        GenAIAttributes.GEN_AI_OPERATION_NAME: operation_name,
+        GenAIAttributes.GEN_AI_SYSTEM: GenAIAttributes.GenAiSystemValues.COHERE.value,
+        GenAIAttributes.GEN_AI_REQUEST_MODEL: kwargs.get("model"),
+        GenAIAttributes.GEN_AI_REQUEST_MAX_TOKENS: kwargs.get("max_tokens"),
+        GenAIAttributes.GEN_AI_REQUEST_FREQUENCY_PENALTY: kwargs.get(
+            "stop_sequences"
+        ),
+        GenAIAttributes.GEN_AI_REQUEST_TEMPERATURE: kwargs.get("temperature"),
+        # TODO: Add to sem conv
+        "gen_ai.cohere.request.seed": kwargs.get("seed"),
+        GenAIAttributes.GEN_AI_REQUEST_PRESENCE_PENALTY: kwargs.get(
+            "presence_penalty"
+        ),
+        GenAIAttributes.GEN_AI_REQUEST_FREQUENCY_PENALTY: kwargs.get(
+            "frequency_penalty"
+        ),
+        GenAIAttributes.GEN_AI_REQUEST_TOP_K: kwargs.get("k"),
+        GenAIAttributes.GEN_AI_REQUEST_TOP_P: kwargs.get("p"),
+    }
+    response_format = kwargs.get("response_format")
+    if response_format:
+        # TODO: Add to sem conv
+        attributes["gen_ai.cohere.request.response_format"] = response_format.type
+
+    set_server_address_and_port(client_instance, attributes)
+
+    # filter out None values
+    return {k: v for k, v in attributes.items() if v is not None}
+
+
+def message_to_event(message: cohere.types.ChatMessageV2, capture_content: bool) -> Event:
+    attributes = {
+        GenAIAttributes.GEN_AI_SYSTEM: GenAIAttributes.GenAiSystemValues.COHERE.value
+    }
+    role = get_property_value(message, "role")
+    content = get_property_value(message, "content")
+
+    body = {}
+    if capture_content and content:
+        body["content"] = content
+    if role == "assistant":
+        tool_calls = extract_tool_calls(message, capture_content)
+        if tool_calls:
+            body = {"tool_calls": tool_calls}
+    elif role == "tool":
+        tool_call_id = get_property_value(message, "tool_call_id")
+        if tool_call_id:
+            body["id"] = tool_call_id
+
+    return Event(
+        name=f"gen_ai.{role}.message",
+        attributes=attributes,
+        body=body if body else None,
+    )
+
+
+def set_response_attributes(
+    span: Span, result: cohere.ChatResponse, event_logger: EventLogger, capture_content: bool
+):
+    event_logger.emit(_response_to_event(result, capture_content))
+
+    span.set_attribute(
+        GenAIAttributes.GEN_AI_RESPONSE_FINISH_REASONS,
+        [result.finish_reason],
+    )
+
+    if getattr(result, "id", None):
+        span.set_attribute(GenAIAttributes.GEN_AI_RESPONSE_ID, result.id)
+
+    # Get the usage
+    if getattr(result, "usage", None):
+        if getattr(result.usage, "tokens"):
+            span.set_attribute(
+                GenAIAttributes.GEN_AI_USAGE_INPUT_TOKENS,
+                result.usage.tokens.input_tokens,
```

Comment:
PTAL at the https://github.com/open-telemetry/semantic-conventions/issues/1279

Cohere reports billed tokens in addition to input/output 

<img width="647" alt="image" src="https://github.com/user-attachments/assets/79023958-22d0-4790-8df0-cc865a5b62e9" />

https://docs.cohere.com/v2/reference/chat#response.body.usage

I'm not sure what's the difference and if we need to report one or another (or maybe both?)